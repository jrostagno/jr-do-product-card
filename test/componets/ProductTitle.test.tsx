import React from 'react';

import { render } from '@testing-library/react';
import { ProductTitle } from '../../src/components/ProductTitle';
import { ProductCard } from '../../src/components/ProductCard';
import { product1 } from '../data/product';

describe('ProductTitle', () => {
  test('Debe mostrar el componente con el titulo personalizado', () => {
    const { asFragment } = render(
      <ProductTitle title="Custom Product" className="custom-title" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('Debe de mostrar el componente con el nombre del producto', () => {
    const product = render(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(product.rerender).toMatchSnapshot();
  });
});
