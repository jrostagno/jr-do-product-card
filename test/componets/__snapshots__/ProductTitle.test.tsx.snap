// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de mostrar el componente con el nombre del producto 1`] = `[Function]`;

exports[`ProductTitle Debe mostrar el componente con el titulo personalizado 1`] = `
<DocumentFragment>
  <span
    class="productDescription custom-title"
  >
    Custom Product
  </span>
</DocumentFragment>
`;
